examples/templates/ecs-container/main.tf

```diff
@@ -0,0 +1,242 @@
+}
+
+locals {
+
+  # User data is used to stop/start AWS instances. See:
+  # https://github.com/hashicorp/terraform-provider-aws/issues/22
+
+  user_data_start = <<EOT
+Content-Type: multipart/mixed; boundary="//"
+MIME-Version: 1.0
+
+--//
+Content-Type: text/cloud-config; charset="us-ascii"
+MIME-Version: 1.0
+Content-Transfer-Encoding: 7bit
+Content-Disposition: attachment; filename="cloud-config.txt"
+
+#cloud-config
+cloud_final_modules:
+- [scripts-user, always]
+hostname: ${lower(data.coder_workspace.me.name)}
+
+--//
+Content-Type: text/x-shellscript; charset="us-ascii"
+MIME-Version: 1.0
+Content-Transfer-Encoding: 7bit
+Content-Disposition: attachment; filename="userdata.txt"
+
+#!/bin/bash
+sudo -u ubuntu sh -c '${coder_agent.coder.init_script}'
+--//--
+EOT
+
+  user_data_end = <<EOT
+Content-Type: multipart/mixed; boundary="//"
+MIME-Version: 1.0
+--//
+Content-Type: text/cloud-config; charset="us-ascii"
+MIME-Version: 1.0
+Content-Transfer-Encoding: 7bit
+Content-Disposition: attachment; filename="cloud-config.txt"
+#cloud-config
+cloud_final_modules:
+- [scripts-user, always]
+--//
+Content-Type: text/x-shellscript; charset="us-ascii"
+MIME-Version: 1.0
+Content-Transfer-Encoding: 7bit
+Content-Disposition: attachment; filename="userdata.txt"
+#!/bin/bash
+sudo shutdown -h now
+--//--
+EOT
+}
+
+# user data definitions, for both ECS & Coder
+data "cloudinit_config" "main" {
+  base64_encode = true
+  gzip          = false
+
+  part {
+    content = data.coder_workspace.me.transition == "start" ? local.user_data_start : local.user_data_end
+  }
+  # required for the EC2 instances to join the ECS cluster
+  part {
+    content = "#!/bin/bash\necho ECS_CLUSTER=${aws_ecs_cluster.main.name} >> /etc/ecs/ecs.config"
+  }
+}
+
+# set launch template for EC2 auto-scaling group
+resource "aws_launch_template" "coder-oss-ubuntu" {
+  name_prefix = "coder-oss"
+  # ECS optimized AMI - https://github.com/awsdocs/amazon-ecs-developer-guide/blob/master/doc_source/ecs-optimized_AMI.md
+  image_id      = "ami-03f8a7b55051ae0d4"
+  instance_type = "t2.medium"
+  iam_instance_profile {
+    arn = "arn:aws:iam::816024705881:instance-profile/coder-oss-ecs-role"
+  }
+  user_data = data.cloudinit_config.main.rendered
+}
+
+# provision auto-scaling group to host ECS task definitions
+resource "aws_autoscaling_group" "main" {
+  name                  = "coder-ecs-auto-scaling-group"
+  min_size              = 1
+  max_size              = 1
+  desired_capacity      = 1
+  availability_zones    = ["us-east-1a"]
+  protect_from_scale_in = true
+
+  launch_template {
+    id      = aws_launch_template.coder-oss-ubuntu.id
+    version = "$Latest"
+  }
+
+  tag {
+    key                 = "AmazonECSManaged"
+    value               = true
+    propagate_at_launch = true
+  }
+}
+
+# create AWS ECS cluster
+resource "aws_ecs_cluster" "main" {
+  name = "coder-oss-ecs"
+
+  setting {
+    name  = "containerInsights"
+    value = "enabled"
+  }
+}
```

Comment:
Interestingly, this didn't create a new cluster since `coder-oss-ecs` already exists in our AWS account from your tests. It just added the service to the cluster and

Terraform did notify me that it detected drift in the build logs. Here's the resulting cluster.

![cluster](https://user-images.githubusercontent.com/22407953/188751989-148104b9-8052-4aec-8b7b-0c75fdfcff98.png)

If you want 1 cluster per workspace, you could change the name to include the workspace ID like we did for the rest.